File: client/javascript-closure-angular/API/Client/SiteList.js
goog.provide('API.Client.SiteList');

/**
 * @record
 */
API.Client.SiteList = function() {}

/**
 * @type {!API.Client.Metadata}
 * @export
 */
API.Client.SiteList.prototype.metadata;

/**
 * @type {!API.Client.Pagination}
 * @export
 */
API.Client.SiteList.prototype.pagination;

/**
 * @type {!Array<!API.Client.Site>}
 * @export
 */
API.Client.SiteList.prototype.data;

File: src/main/scala/org/openapitools/client/api/SiteList.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import SiteList._

case class SiteList (
  metadata: Option[Metadata],
pagination: Option[Pagination],
data: Option[List[Site]])

object SiteList {
  import DateTimeCodecs._

  implicit val SiteListCodecJson: CodecJson[SiteList] = CodecJson.derive[SiteList]
  implicit val SiteListDecoder: EntityDecoder[SiteList] = jsonOf[SiteList]
  implicit val SiteListEncoder: EntityEncoder[SiteList] = jsonEncoderOf[SiteList]
}
